pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/LoadManager.java

```diff
@@ -18,73 +18,105 @@
-     */
-    ResourceUnit getLeastLoaded(ServiceUnitId su) throws Exception;
-
-    /**
-     * Generate the load report
-     */
-    LoadReport generateLoadReport() throws Exception;
-
-    /**
-     * Set flag to force load report update
-     */
-    void setLoadReportForceUpdateFlag();
-
-    /**
-     * Publish the current load report on ZK
-     */
-    void writeLoadReportOnZookeeper() throws Exception;
-
-    /**
-     * Update namespace bundle resource quota on ZK
-     */
-    void writeResourceQuotasToZooKeeper() throws Exception;
-
-    /**
-     * Generate load balancing stats metrics
-     */
-    List<Metrics> getLoadBalancingMetrics();
-
-    /**
-     * Unload a candidate service unit to balance the load
-     */
-    void doLoadShedding();
-
-    /**
-     * Namespace bundle split
-     */
-    void doNamespaceBundleSplit() throws Exception;
-
-    /**
-     * Removes visibility of current broker from loadbalancer list so, other brokers can't redirect any request to this
-     * broker and this broker won't accept new connection requests.
-     *
-     * @throws Exception
-     */
-    public void disableBroker() throws Exception;
-
-    public void stop() throws PulsarServerException;
+	public void start() throws PulsarServerException;
+
+	/**
+	 * Is centralized decision making to assign a new bundle.
+	 */
+	boolean isCentralized();
+
+	/**
+	 * Returns the Least Loaded Resource Unit decided by some algorithm or
+	 * criteria which is implementation specific
+	 */
+	ResourceUnit getLeastLoaded(ServiceUnitId su) throws Exception;
+
+	/**
+	 * Generate the load report
+	 */
+	LoadReport generateLoadReport() throws Exception;
+
+	/**
+	 * Set flag to force load report update
+	 */
+	void setLoadReportForceUpdateFlag();
+
+	/**
+	 * Publish the current load report on ZK
+	 */
+	void writeLoadReportOnZookeeper() throws Exception;
+
+	/**
+	 * Update namespace bundle resource quota on ZK
+	 */
+	void writeResourceQuotasToZooKeeper() throws Exception;
+
+	/**
+	 * Generate load balancing stats metrics
+	 */
+	List<Metrics> getLoadBalancingMetrics();
+
+	/**
+	 * Unload a candidate service unit to balance the load
+	 */
+	void doLoadShedding();
+
+	/**
+	 * Namespace bundle split
+	 */
+	void doNamespaceBundleSplit() throws Exception;
+
+	/**
+	 * Determine the broker root.
+	 */
+	String getBrokerRoot();
+
+	/**
+	 * Removes visibility of current broker from loadbalancer list so, other
+	 * brokers can't redirect any request to this broker and this broker won't
+	 * accept new connection requests.
+	 *
+	 * @throws Exception
+	 */
+	public void disableBroker() throws Exception;
+
+	public void stop() throws PulsarServerException;
+
+	static LoadManager create(final PulsarService pulsar) {
+		try {
+			final ServiceConfiguration conf = pulsar.getConfiguration();
+			final Class<?> loadManagerClass = Class.forName(conf.getLoadManagerClassName());
+			// Assume there is a constructor with one argument of PulsarService.
```

Comment:
That sounds fine, the only problem I would have with that is that I would lose the ability to finalize many fields, but that probably is not a big deal so I will change it